documentation/design/proposals/console-1.0-backend.adoc

```diff
@@ -0,0 +1,117 @@
+
+* MessagingTenant
+* MessagingAddress
+* MessagingEndpoint
+* MessagingUser
+* MessagingRole
+* MessagingRoleBinding
+* IoTProject
+
+Given the timelines, service admin console should not be in scope for 1.0
+
+=== Console API/UI changes
+
+==== AddressSpace => MessagingTenant
+
+image::console_1.0_overview.png[]
+
+* `AddressSpace` is removed and replaced by `MessagingTenant`, and there can now only be a single MessagingTenant named 'default' per Kubernetes namespace. Most of the properties like plan and authentication service (now identity provider) is now optional, and the minimum information needed to create a tenant is its namespace. In addition, a tenant may have a set of capabilities such as 'transactional'.
+
+* Remove customizing endpoints as part of creating `AddressSpace`/`MessagingTenant` (enmasse-operator may create default endpoint specified in a `MessagingPlan`). In 1.0, one or more endpoints may be created independently of the tenant.
+
+==== MessagingEndpoint
+
+A new workflow for creating `MessagingEndpoints` resources is needed. This workflow should allow creating endpoints of new types that are supported in 1.0. Supported types are 'cluster', 'route', 'loadBalancer', 'ingress' and 'nodePort'.
+
+==== MessagingAddress
+
+Requires renaming elements from what we already have from 1.x. Address -> MessagingAddress (though UI can still display Address). In addition, each type of address have a set of additional configuration that should be possible to specify in the UI. There is also a new address type 'deadLetter' supported.
+
+The properties that should be possible to set are (will probably be extended):
+
+* Queues: expiryAddress, deadLetterAddress, partitions, messageTtl min+max, maxConsumers
+* Topics: ...
+* Subscription: topic, maxConsumers, deadLetterAddress, expiryAddress
+* Anycast: ...
+* Multicast: ...
+* DeadLetter: ...
+
+
+==== MessagingUser, MessagingRole and MessagingRoleBinding
+
+A new flow to create a MessagingUser is needed. This resources is based on https://github.com/EnMasseProject/enmasse/blob/master/documentation/design/proposals/authn_authz_for_1.0.adoc#messaginguser .
+
+MessagingRole and MessagingRoleBinding may be created as additional steps in this flow for simplicity, or as a separate flow.
+
+=== Backend changes
+
+==== Architecture 
+
+image::console_1.0_architecture.png[]
+
+The arrows indicate the flow of data.
+
+==== Metrics
+
+At the moment console-server queries connections/links from the address space agents.
+
+For 1.0, the agent disappears, so information must be gathered differently:
+
+* Deploy an EnMasse-owned ephemeral Prometheus instance as planned in the initial console proposal.
+* Configure prometheus to scrape router and broker pods
+* Console-server querying the routers directly to build the list of connections and their links
+* Console server querying the prometheus instance for address and connection metrics
+* Console-server eventually also becomes backend for the CLI.
```

Comment:
I wonder if we should put the exposing the graphql service (cURL with preemptive auth) as part of this release.  This will allow us to test some features directly against the graphql interface rather than writing selenium based tests.    This will allow us to reduce the number of slow selenium tests (slow both in terms of their writing and running).  We'll still need to make sure the whole UI is covered, but we should be able to do this with fewer tests.